src/dask_awkward/lib/core.py

```diff
@@ -568,7 +572,20 @@ def repartition(self, npartitions=None, divisions=None, rows_per_partition=None)
         if rows_per_partition:
             divisions = list(range(0, nrows, rows_per_partition))
             divisions.append(nrows)
-        return repartition_layer(self, divisions)
+
+        new_layer = repartition_layer(self, key, divisions)
+        new_layer = MaterializedLayer(new_layer)
+        new_layer._opt_mock_materialized = True
```

Comment:
Survives pickle?

Reply:
I'm not sure, we've do this elsewhere with `_opt_touch_all` which was the workaround introduced to make sure `dak.zip` would work.